runtime/interpreter/interpreter_expression.go

```diff
@@ -443,7 +443,11 @@ func (interpreter *Interpreter) VisitIntegerExpression(expression *ast.IntegerEx
 	value := expression.Value
 
 	if _, ok := typ.(*sema.AddressType); ok {
-		return NewAddressValueFromBytes(value.Bytes())
+		bytes := value.Bytes()
+		return NewAddressValueFromBytes(
+			interpreter,
+			bytes,
+		)
```

Comment:
`value.Bytes()` allocates new memory. Must delay call to it, and make the constructor `NewAddressValueFromBytes` accept an anon-func.
```suggestion
		return NewAddressValueFromBytes(
			interpreter,
			value.Bytes,
		)
```